libs/pando/engine/src/optimization/prune.spec.ts

```
@@ -0,0 +1,348 @@
+          const m3 = new State([flip(n3, inc)], builds, 'q').monotonicities
+          expect(m3.get('c0')).toEqual({ inc, dec: !inc })
+          expect(m3.get('c1')).toEqual({ inc, dec: !inc })
+          expect(m3.get('c2')).toEqual({ inc, dec: !inc })
+        })
+    })
+    describe('maybe-zero prod', () => {
+      for (const inc of [true, false])
+        test(inc ? 'inc' : 'dec', () => {
+          const n1 = prod(sum(r0, -7), r1, r2)
+          const m1 = new State([flip(n1, inc)], builds, 'q').monotonicities
+          expect(m1.get('c0')![inc ? 'dec' : 'inc']).toEqual(false)
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)